apps/chat/src/utils/app/publications.ts

```diff
@@ -0,0 +1,31 @@
+import { ApiKeys } from '@/src/types/common';
+import { SharingType } from '@/src/types/share';
+
+import { constructPath } from './file';
+
+export const isItemPublic = (id: string) => id.split('/')[1] === 'public';
+
+export const createTargetUrl = (
+  apiKey: ApiKeys,
```

Comment:
Use `featureType: FeatureType` instead
```suggestion
  featureType: FeatureType,
```
and `getApiKeyByFeatureType` on service level.
Avoid to use `ApiKeys` in our logic